applications/dashboard/models/class.spammodel.php

```diff
@@ -110,9 +110,99 @@ public static function isSpam($RecordType, $Data, $Options = array()) {
                     break;
             }
 
-            LogModel::insert('Spam', $RecordType, $Data, $LogOptions);
+            // If this is a discussion or a comment, it needs some special handling.
+            if ($RecordType == 'Comment' || $RecordType == 'Discussion') {
+                // Grab the record ID, if available.
+                switch ($RecordType) {
+                    case 'Comment':
+                        $recordID = intval(val('CommentID', $Data, false));
+                        break;
+                    case 'Discussion':
+                        $recordID = intval(val('DiscussionID', $Data, false));
+                        break;
+                }
+
+                /**
+                 * If we have a valid record ID, run it through flagForReview.  This will allow us to purge existing
+                 * discussions and comments that have been flagged as SPAM after being edited.  If there's no valid ID,
+                 * just treat it with regular SPAM logging.
+                 */
+                if ($recordID) {
+                    self::flagForReview($RecordType, $recordID, $Data);
+                } else {
+                    LogModel::insert('Spam', $RecordType, $Data, $LogOptions);
+                }
+            } else {
+                LogModel::insert('Spam', $RecordType, $Data, $LogOptions);
+            }
         }
 
         return $Spam;
     }
+
+    /**
+     * Insert a SPAM Queue entry for the specified record and delete the record, if possible.
+     *
+     * @param string $recordType The type of record we're flagging: Discussion or Comment.
+     * @param int $id ID of the record we're flagging.
+     * @param object|array $data Properties used for updating/overriding the record's current values.
+     *
+     * @throws Exception If an invalid record type is specified, throw an exception.
+     */
+    protected static function flagForReview($recordType, $id, $data) {
+        // We're planning to purge the spammy record.
+        $deleteRow = true;
+
+        /**
+         * We're only handling two types of content: discussions and comments.  Both need some special setup.
+         * Error out if we're not dealing with a discussion or comment.
+         */
+        switch ($recordType) {
+            case 'Comment':
+                $model = new CommentModel();
+                $row = $model->getID($id, DATASET_TYPE_ARRAY);
+                break;
+            case 'Discussion':
+                $model = new DiscussionModel();
+                $row = (array)$model->getID($id);
```

Comment:
Can we pick either array-casting or using `DATASET_TYPE_ARRAY` and stick with it?